libuhd: update to 4.8.0.0

chrisjohgorman opened the pull request:
- Removed use of cast of thrd->native_handle() to (pthread_t) in uhd::set_thread_name opting for no thread naming on windows.  This follows gnuradio which doesn't name threads on windows.
- Add new dependencies and makedepends for package.
- Renamed and rebuilt patch against new uhd sources.
- Rebuild gnuradio against new libuhd.

ognevny commented on mingw-w64-libuhd/PKGBUILD:
```diff
@@ -67,12 +71,18 @@ build() {
   DESTDIR="${pkgdir}" "${MINGW_PREFIX}"/bin/cmake.exe --install "build-${MSYSTEM}"
 
+  cd "${srcdir}"/"build-${MSYSTEM}"/"python"
+  pip install --ignore-installed --prefix="${pkgdir}"/"${MINGW_PREFIX}" dist/uhd*.whl
```
is it possible to implement this with `python-install`? https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-PKGBUILD-templates%2FPKGBUILD.python.sh#L53

ognevny commented on mingw-w64-libuhd/PKGBUILD:
```diff
@@ -18,23 +18,27 @@ depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs"
 
-  patch -Np1 -i "${srcdir}"/0001-libraries_headers_and_pthread_setname_np_cast.patch
+  patch -Np1 -i "${srcdir}"/0001-libraries_headers_and_no_pthread_setname_np.patch
+
```
unneded newline :)

chrisjohgorman replied:
Hi and than you @ognevny.  I have added the requested changes.  I am keeping this as a work in progress as the build machine artifacts did not run properly on my machine.  I hope with the suggested changes they will run properly after the build.  (My local builds ran fine, but the build machine artifacts couldn't load a library properly.)

ognevny replied:
they probably won't resolve issue. that was nits to better match common PKGBUILD style

chrisjohgorman commented:
Well @ognevny you were correct, it didn't fix the problem.  I think I have
narrowed it down.  The build machine links to python312.dll and my machine
links to libpython3.12.dll.  Off to see why.

On Tue, Apr 1, 2025 at 4:24 PM Maksim Bondarenkov ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In mingw-w64-libuhd/PKGBUILD
> <https://github.com/msys2/MINGW-packages/pull/23831#discussion_r2023626776>
> :
>
> >
>  prepare() {
>    cd "${srcdir}"/uhd-${pkgver}
>
> -  patch -Np1 -i "${srcdir}"/0001-libraries_headers_and_pthread_setname_np_cast.patch
> +  patch -Np1 -i "${srcdir}"/0001-libraries_headers_and_no_pthread_setname_np.patch
> +
>
> they probably won't resolve issue :( that was nits to better match common
> PKGBUILD style
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/msys2/MINGW-packages/pull/23831#discussion_r2023626776>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG76X44IJSKCOE3HZDPSGKL2XLYWDAVCNFSM6AAAAAB2FNGSDKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDOMZTHE3DIOJUGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

chrisjohgorman commented:
A rewrite of one of the points in my git commit with typos corrected.
- add some guard code to UHDPython.cmake to ensure that cmake's find_package()
  command doesn't use the libraries in the buildmachine's c:/hostedtoolcache
  directory.

This looks good to go all packages install and the library link bug has been fixed.

chrisjohgorman commented:
I've made all of @ognevny 's recommended changes, and fixed the linking error made by cmake's find_package(Python3) call.  I've squashed my 3 commits into 1 and run.  I don't own any Ettus Research USRPs, so I can't test functionality of the updated software on real hardware, so I tested the ability of python to import the uhd module and ran the test cases.  For the test cases I get 94% passing on ucrt64, 96% passing on mingw64 and 98% passing on clang64.  I get two test failures and between 0 and 4 test timeouts depending on which MINGW_ARCH the tests run under. 

I believe this is ready for inclusion, but if there are more changes to be done, please let me know.
